### DISCUSSION TITLE
Does Next.js uses SWR library inbuilt or needs to integrate separately?

### DISCUSSION THREAD

**Discussion by jaimeenkoffeekodes** [vercel/next.js#66782] [Help]
### Summary

Doubt regarding SWR!

**Comment by icyJoseph** [comment]
`swr` is a data fetching hook, which you have to install yourself. Is there anything in particular that lead you to think this?

SWR though is also a technique, Stale-While-Revalidate, https://datatracker.ietf.org/doc/html/rfc5861, which basically says, serve any data we have, and after a certain time, revalidate it, but never make any clients wait for data. 

> If a cached response is served stale due to the presence of this
   extension, the cache SHOULD attempt to revalidate it while still
   serving stale responses (i.e., without blocking).

And that concept, as such, is indeed used in a couple of places.
